### PR TITLE
Convert twitter universal website tag code to third party integration

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

Updated: This PR updates the  Twitter universal website tag code previously added inline within `base.scala.html` in this PR: https://github.com/guardian/frontend/pull/20876 to be a third party integration. 

The original implementation (inline within `base.scala.html`) was only agreed on the basis of this being a temporary addition, now that it's required beyond the initial period of it's use then an integration as per @katebee's comment https://github.com/guardian/frontend/pull/20876#discussion_r243815910 is the correct way to load this script, and this can now also include a check against whether the user has granted consent to personalised ads & tracking in the `shouldRun` check.

I've also put a switch in so we can turn this tracking script on/off at any given time.

## Screenshots

N/A

## What is the value of this and can you measure success?

Makes sure we're no longer firing tracking when consent hasn't been granted

### CODE REVIEW

**Comment by GHaberis** [comment]
FAO @guardian/dotcom-platform @rcrphillips

**Comment by GHaberis** [comment]
Update: @rcrphillips and I have discussed this and marketing still require the tracking so I'm going to implement this as a third party tag behind a consent check.

**Comment by rcrphillips** [comment]
Having spoken to the marketing team, they have recognised the value of this beyond the campaign that they ran in January. Thus, if we were able to move it to a more permanent location, that would be great.

File: common/app/conf/switches/CommercialSwitches.scala [hunk 1]
```diff
@@ -98,6 +98,16 @@ trait CommercialSwitches {
     exposeClientSide = true
   )
 
+  val TwitterUwtSwitch = Switch(
```
Comment (ripecosta) [inline common/app/conf/switches/CommercialSwitches.scala]: Great idea adding a switch 🥇

File: static/src/javascripts/projects/commercial/modules/third-party-tags/twitter-uwt.spec.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import { getAdConsentState as _getAdConsentState } from 'common/modules/commercial/ad-prefs.lib';
+
+const getAdConsentState: any = _getAdConsentState;
+
+jest.mock('common/modules/commercial/ad-prefs.lib', () => ({
+    getAdConsentState: jest.fn(),
+}));
+
+/**
+ * we have to mock config like this because
```
Comment (ripecosta) [inline static/src/javascripts/projects/commercial/modules/third-party-tags/twitter-uwt.spec.js]: Since twitterUwt returns a function and not an object like most third party tag modules we can test the 'regular' way.

File: static/src/javascripts/projects/commercial/modules/third-party-tags/twitter-uwt.spec.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    it('shouldRun to be false if ad consent not granted', () => {
+        getAdConsentState.mockReturnValueOnce(false);
+
+        const { shouldRun, url, onLoad } = twitterUwt();
+
+        expect(shouldRun).toEqual(false);
+        expect(url).toEqual('//static.ads-twitter.com/uwt.js');
+        expect(onLoad).toBeDefined();
+    });
+});
```
Comment (ripecosta) [inline static/src/javascripts/projects/commercial/modules/third-party-tags/twitter-uwt.spec.js]: We should also have a test for when getAdConsentState returns null, to have the tests be as complete as possible.

**Review by ripecosta — APPROVED** [review]
Added a comments to the tests but to be honest those are non blocking at this point as we'll have to change all the tests again when the CMP library is released.

File: static/src/javascripts/projects/commercial/modules/third-party-tags/twitter-uwt.spec.js (hunk 3 above)
Comment (GHaberis) [reply static/src/javascripts/projects/commercial/modules/third-party-tags/twitter-uwt.spec.js]: Good suggestion, I've now added this test!
